File: js/2020/day1/solve.js
let data = prompt("input");
let nums = data.split("\n").map(Number);

function solveSecond() {
    for (let i = 0; i < nums.length; i++) {
        for (let j = i; j < nums.length; j++) {
            for (let k = j; k < nums.length; k++) {
                let x = nums[i];
                let y = nums[j];
                let z = nums[k];
                if (+(x+y+z) === 2020) {
                    return x*y*z;
                }
            }
        }
    }
    return null;
}

function solveFirst() {
    nums.sort();
    let done = false;
    let small = 0;
    let large = nums.length - 1;
    while (!done) {
        sum = nums[small] + nums[large];
        if (sum > 2020)
            large--;
        if (sum < 2020)
            small++;
        
        if (sum === 2020) {
            done = true;
            return nums[small] * nums[large];
        }
    }
}

console.log(solveFirst());
console.log(solveSecond());

File: python/2020/day1/solve.py
def solve(data):
    ans1 = 0
    ans2 = 0
    for x in range(0, len(data)): 
        for y in range(x,len(data)):
            i = data[x]
            j = data[y]
            if ((i+j) == 2020):
                ans1 = i*j
                break
            for z in range(y, len(data)):
                k = data[z]
                if((i+j+k) == 2020):
                    ans2 = i*j*k
                    break
    return ans1,ans2

with open("in.txt") as input:
    nums = []
    for row in input:
        nums.append(int(row))
    nums.sort()
    ans = solve(nums)
    print(ans)
